### DISCUSSION TITLE
Hooks for email transporters

### DISCUSSION THREAD

**Discussion by smilledge** [directus/directus#14049] [Feature Requests]
## Problem

Currently there doesn't seem to be a way to use a [custom email transporter](https://nodemailer.com/plugins/create/#transports) or install [nodemailer plugins](https://nodemailer.com/plugins/).

### mailer.before

This would be used to create a custom transporter. If a transporter is returned then the standard transporter would not be created.

```javascript
register.init('mailer.before', () => {
    return {
        name: 'cool-custom-mailer',
        version: '1.1.1',
        send: () => {
            // Send email...
        },
        ...
    }
});
```

### mailer.after

This would be used to add plugins to the standard transporters.

```javascript
register.init('mailer.after', ({ transporter }) => {
    transporter.use('compile', htmlToText(options))
});
```

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
